ports/servoshell/egl/app.rs

```diff
@@ -258,7 +253,12 @@ pub(crate) struct AppInitOptions {
 
 pub struct App {
     state: Rc<RunningAppState>,
-    platform_window: Rc<EmbeddedPlatformWindow>,
+    // todo: multi-window support, like desktop version.
+    // This is just an intermediate state, to split refactoring into
+    // multiple PRs.
+    // todo: active window?
```

Comment:
Nit: typically `TODO` is spelled with capital letters in Servo, which really only matters when grepping.

The `RunningAppState` stores a list of windows which can give access to the `PlatformWindow` as well. Maybe that's enough here or do we need the actual type of the `PlatformWindow` implementation? I think this is stored as the `trait` in the desktop version. It might be nice to keep the same behavior between ports.

Reply:
> Maybe that's enough here or do we need the actual type of the PlatformWindow implementation?

I think the trait currently doesn't offer access to the HostCallbacks / HostTrait, which are stored in the EmbeddedPlatformWindow. The desktop version doesn't have those, I guess because it assumes winit?

Reply:
I think we could get around this issue by accepting `Rc<HostTrait>` instead of `Box<HostTraits>` and storing it in both the `PlatformWindow` and the `App`. In fact we will have to do something like this if we want to support multiple windows for the embedded platform.